src/cmd/tools/dtest/docker/harness/resources/common.go

```diff
@@ -72,6 +73,14 @@ func (o dockerResourceOptions) withDefaults(
 		o.containerName = defaultOpts.containerName
 	}
 
+	if len(o.image.name) == 0 {
+		o.image.name = defaultOpts.image.name
+	}
+
+	if len(o.image.tag) == 0 {
+		o.image.tag = defaultOpts.image.tag
+	}
```

Comment:
I think these should always be set together. Perhaps worth merging these `if`s into one `if o.image == dockerImage{}`.